terraform/eval_state.go

```diff
@@ -221,15 +355,14 @@ func (n *EvalUndeposeState) Eval(ctx EvalContext) (interface{}, error) {
 		return nil, nil
 	}
 
-	// If we don't have any tainted, then we don't have anything to do
-	if len(rs.Tainted) == 0 {
+	// If we don't have any desposed resource, then we don't have anything to do
+	if len(rs.Deposed) == 0 {
 		return nil, nil
 	}
 
-	// Undepose to the tainted
-	idx := len(rs.Tainted) - 1
-	rs.Primary = rs.Tainted[idx]
-	rs.Tainted[idx] = nil
+	// Undepose
+	rs.Primary = rs.Deposed[len(rs.Deposed)-1]
+	rs.Deposed = rs.Deposed[:len(rs.Deposed)-1]
```

Comment:
Ah yeah, that makes sense. I think this was an artifact of Deposed first being a singleton and then getting flipped to a list. Fixing...